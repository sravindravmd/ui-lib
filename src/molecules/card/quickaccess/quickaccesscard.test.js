import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import QuickAccessCard from './index'

afterEach(cleanup)

describe('<QuickAccessCard/>', () => {
  const imgUrl =
    'https://images.unsplash.com/photo-1603881568692-1429786f5ce1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60'
  it('renders QuickAccessCard correctly ', () => {
    const { container } = render(
      <QuickAccessCard
        title='23987532950 - The Fleet Fox'
        subtitle='Shirts/Flanne'
        imgUrl={imgUrl}
        imgAltText='card-img'
      />
    )
    expect(container).toMatchSnapshot()
  })
  it('should render QuickAccessCard after click', () => {
    const { container, getByText } = render(
      <QuickAccessCard
        title='23987532950 - The Fleet Fox'
        subtitle='Shirts/Flanne'
        imgUrl={imgUrl}
        imgAltText='card-img'
        onClick={() => {}}
      />
    )
    fireEvent.click(getByText('23987532950 - The Fleet Fox'))
    expect(container).toMatchSnapshot()
  })
})
