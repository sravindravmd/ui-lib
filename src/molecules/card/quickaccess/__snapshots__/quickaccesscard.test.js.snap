// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuickAccessCard/> renders QuickAccessCard correctly  1`] = `
<div>
  <div
    class="sc-bdfBwQ kjmIBZ"
  >
    <div
      class="sc-jSgupP leSHHO"
      tabindex="-1"
    >
      <div
        class="sc-gsTCUz cSuJme"
      >
        <img
          alt="card-img"
          src="https://images.unsplash.com/photo-1603881568692-1429786f5ce1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
        />
      </div>
      <div
        class="sc-dlfnbm bEflta"
      >
        <div
          class="sc-hKgILt cOJQlc"
        >
          23987532950 - The Fleet Fox
        </div>
        <div
          class="sc-eCssSg gzwAIZ"
        >
          Shirts/Flanne
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<QuickAccessCard/> should render QuickAccessCard after click 1`] = `
<div>
  <div
    class="sc-bdfBwQ eBWOhk"
  >
    <div
      class="sc-jSgupP leSHHO"
      tabindex="-1"
    >
      <div
        class="sc-gsTCUz cSuJme"
      >
        <img
          alt="card-img"
          src="https://images.unsplash.com/photo-1603881568692-1429786f5ce1?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=500&q=60"
        />
      </div>
      <div
        class="sc-dlfnbm bEflta"
      >
        <div
          class="sc-hKgILt cOJQlc"
        >
          23987532950 - The Fleet Fox
        </div>
        <div
          class="sc-eCssSg gzwAIZ"
        >
          Shirts/Flanne
        </div>
      </div>
    </div>
  </div>
</div>
`;
